src/codecs/tga/decoder.rs

```diff
@@ -446,14 +450,23 @@ impl<'a, R: 'a + Read + Seek> ImageDecoder<'a> for TgaDecoder<R> {
                 &buf[..num_raw_bytes]
             } else {
                 fallback_buf.resize(num_raw_bytes, 0u8);
-                self.r.by_ref().read_exact(&mut fallback_buf[..num_raw_bytes])?;
+                self.r
+                    .by_ref()
+                    .read_exact(&mut fallback_buf[..num_raw_bytes])?;
                 &fallback_buf[..num_raw_bytes]
             }
         };
 
         // expand the indices using the color map if necessary
         if self.image_type.is_color_mapped() {
             let pixel_data = self.expand_color_map(rawbuf)?;
+            let difference = pixel_data.len() as i32 - buf.len() as i32;
+            // not enough data to fill the buffer, or would overflow the buffer
+            if difference != 0 {
```

Comment:
Is there a reason this isn't `if pixel_data.len() != buf.len() { ... }`?

Reply:
Not really. Just thought it might be useful to know which one of the two buffers is the bigger one in case you're able to resize one of them to make it fit into the other. I definitely see now how it can be confusing, do you want me to change it?

Reply:
Yeah, the equality check would be preferable. Casting from `usize` -> `i32` can overflow on 64-bit systems

Reply:
No problem! I changed it.